server/src/main/java/org/elasticsearch/health/node/selection/HealthNodeSelector.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node.selection;
+
+import org.elasticsearch.persistent.AllocatedPersistentTask;
+import org.elasticsearch.tasks.TaskId;
+
+import java.util.Map;
+
+/**
+ * Main component used for selecting the health node of the cluster
+ */
+public class HealthNodeSelector extends AllocatedPersistentTask {
+
+    public static final boolean FEATURE_FLAG_ENABLED = "true".equals(System.getProperty("es.health_node_feature_flag_enabled"));
+
+    public static boolean isEnabled() {
+        return FEATURE_FLAG_ENABLED;
+    }
+
+    public static final String TASK_NAME = "health-node-selector";
```

Comment:
Should the task name be "health-node" instead, rather than the task being the node selector (since the task node is the selected node)?

Reply:
Hm, not sure. I thought the name of the task should explain what the task does. The way I see it the task selects the health node, like the `geoip-downloader` downloads the geoip databases, right? If we would name it `health-node` it feels to me like we are saying that the task is the health node... 

I do not have a strong opinion about this, but I do want to discuss it so I can better understand the naming conventions.

Reply:
I get it now, I will change it.